test-clients/src/main/java/com/hedera/services/bdd/suites/token/TokenUpdateSpecs.java

```diff
@@ -587,6 +593,82 @@ public HapiApiSpec updateHappyPath() {
 				);
 	}
 
+	public HapiApiSpec updateTokenTreasuryHappyPath() {
+		return defaultHapiSpec("UpdateTokenTreasuryHappyPath")
+				.given(
+						cryptoCreate(TOKEN_TREASURY),
+						cryptoCreate("newTokenTreasury"),
+						newKeyNamed("adminKey"),
+						newKeyNamed("supplyKey"),
+						tokenCreate("primary")
+								.treasury(TOKEN_TREASURY)
+								.initialSupply(500)
+								.adminKey("adminKey")
+								.supplyKey("supplyKey")
+				)
+				.when(
+						tokenAssociate("newTokenTreasury", "primary"),
+						tokenUpdate("primary")
+								.treasury("newTokenTreasury").via("tokenUpdateTxn")
+				)
+				.then(
+						getAccountBalance(TOKEN_TREASURY)
+								.hasTokenBalance("primary", 0),
+						getAccountBalance("newTokenTreasury")
+								.hasTokenBalance("primary", 500),
+						getTokenInfo("primary")
+								.hasTreasury("newTokenTreasury")
+								.hasTotalSupply(500),
+						getTxnRecord("tokenUpdateTxn").logged()
+				);
+	}
+
+	public HapiApiSpec updateNftTreasuryHappyPath() {
+		return defaultHapiSpec("UpdateNftTreasuryHappyPath")
+				.given(
+						cryptoCreate(TOKEN_TREASURY),
+						cryptoCreate("newTokenTreasury"),
+						newKeyNamed("adminKeyA"),
+						newKeyNamed("adminKeyB"),
+						newKeyNamed("supplyKeyA"),
+						newKeyNamed("supplyKeyB"),
+						tokenCreate("primary")
+								.tokenType(TokenType.NON_FUNGIBLE_UNIQUE)
+								.treasury(TOKEN_TREASURY)
+								.initialSupply(0)
+								.adminKey("adminKeyA")
+								.supplyKey("supplyKeyA"),
+						tokenCreate("secondary")
+								.treasury("newTokenTreasury")
+								.initialSupply(0)
+								.adminKey("adminKeyB")
+								.supplyKey("supplyKeyB"),
+						mintToken("primary", List.of(ByteString.copyFromUtf8("memo1")))
+				)
+				.when(
+						tokenAssociate("newTokenTreasury", "primary"),
+						tokenUpdate("primary")
+								.treasury("newTokenTreasury").via("tokenUpdateTxn")
+				)
+				.then(
+						getAccountBalance(TOKEN_TREASURY)
+								.hasTokenBalance("primary", 0),
+						getAccountBalance("newTokenTreasury")
+								.hasTokenBalance("primary", 1),
+						getTokenInfo("primary")
+								.hasTreasury("newTokenTreasury")
+								.logged(),
+						getTokenNftInfo("primary", 1)
+								.hasAccountID(TOKEN_TREASURY) // still the same treasury since no token transfer is initiated
+								.logged(),
+						getTokenNftInfos("primary", 0, 1)
+								.hasNfts(
+										newTokenNftInfo("primary", 1, TOKEN_TREASURY, ByteString.copyFromUtf8("memo1"))
+								) // still the same treasury since no token transfer is initiated
+								.logged(),
```

Comment:
This is should not happen (returning TOKEN_TREASURY instead of newTokenTreasury)